### PR TITLE
handle Processing instructions in SimpleXmlParser

### LINKED ISSUE #780: SimpleXmlParser: Processing instructions are not handled
Permitted states do not include [processing instructions](https://www.w3.org/TR/xml/#sec-pi):

https://github.com/oracle/helidon-build-tools/blob/d216b571bb03abd5b08f5a6c65077892e373a871/common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java#L226-L239

The manifesting symptom is that in a processing instruction like this:
```xml
<?m2e execute onConfiguration,onIncremental?>
```
…the parser bails out here:

https://github.com/oracle/helidon-build-tools/blob/d216b571bb03abd5b08f5a6c65077892e373a871/common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java#L593-L594

### PR DESCRIPTION
**Pull request by aserkes**
handle Processing instructions in SimpleXmlParser

Fixes #780 

Signed-off-by: aserkes <andrii.serkes@oracle.com>

### CODE REVIEW

**Review by romain-grecourt — APPROVED** [review]

File: common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java [hunk 1]
```diff
@@ -221,6 +221,15 @@ default List<String> readAttributeList(String name, String qName, Map<String, St
             Collections.addAll(values, value.split(","));
             return values;
         }
+
+        /**
+         * Receive notification of content data inside a Processing instructions element.
```
Comment (ljnelson) [inline common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java:226]: "Processing instructions" -> "processing instruction"

File: common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java [hunk 2]
```diff
@@ -221,6 +221,15 @@ default List<String> readAttributeList(String name, String qName, Map<String, St
...
+         *
+         * @param data the content data of Processing instructions
```
Comment (ljnelson) [inline common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java:228]: "Processing instructions" -> "a processing instruction"

File: common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java [hunk 3]
```diff
@@ -221,6 +221,15 @@ default List<String> readAttributeList(String name, String qName, Map<String, St
...
+         * @throws XMLReaderException if any error occurs
+         */
+        default void processInstructionContent(String data) {
```
[the remark this answers is not in the document]
Comment (romain-grecourt) [reply common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java:231]: Good point. Seems like a typo given the context and also it's an event so using the verb "process" does not make sense.

The method should probably also include the instruction name.
How about something like this?
```java
default void processingInstruction(String name, String data) {
}
```

[the remark this answers is not in the document]
Comment (romain-grecourt) [reply common/xml/src/main/java/io/helidon/build/common/xml/SimpleXMLParser.java:231]: Reading https://www.w3.org/TR/REC-xml/#sec-pi "target" might be better "name" ; though I've seen "name" used while googling.

The target "xml" is reserved though I don't think we want to be strict and enforce that.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**